Neos.ContentRepositoryRegistry/Classes/Service/AsynchronousCatchUpRunnerState.php

```diff
@@ -0,0 +1,60 @@
+<?php
+namespace Neos\ContentRepositoryRegistry\Service;
+
+use Neos\Cache\Frontend\FrontendInterface;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\Projection\ProjectionInterface;
+
+/**
+ *
+ */
+final readonly class AsynchronousCatchUpRunnerState
```

Comment:
I would suggest to skip the `Asynchronous` prefix since this central coordinator is also relevant for _synchronous_ catch-ups.
For example: I would consider switching my installations to sync mode (like @nezaniel did in his spaceship *g) for an additional performance boost. But with concurrent requests this could lead to the infamous "Failed to acquire checkpoint lock for subscriber '...' because it is acquired already" exception.


That again makes me think: Can we outsource some of the `SubprocessProjectionCatchUpTrigger` code so it can be reused for the sync mode (or others potentially) too?